src/Fuzz/Internal.elm

```diff
@@ -1,83 +1,121 @@
-module Fuzz.Internal exposing (Fuzz(..), Fuzzer(Fuzzer), invalidReason, unpackGenTree, unpackGenVal)
+module Fuzz.Internal exposing (Fuzzer, Valid, ValidFuzzer, andThen, andThenNoHistory, combineValid, invalidReason, map)
 
-import Random.Pcg exposing (Generator)
-import RoseTree exposing (RoseTree)
+import Lazy.List exposing ((:::), LazyList)
+import Random.Pcg as Random exposing (Generator)
+import RoseTree exposing (RoseTree(Rose))
 
 
-{- Fuzzers as opt-in RoseTrees
+type alias Fuzzer a =
+    Valid (ValidFuzzer a)
```

Comment:
@rtfeldman Are there downsides of having this as an alias instead of an opaque type, even after it's been aliased again in the not-internal module?